### PR TITLE
Configuration ring and datomic

### PR DESCRIPTION
**Pull request by k-batista**
- Creating a Handler for Exporting Apis
- Adding a datomic connection
- Creating a file to upload csv files to database

- Files: Salary and BFB

### CODE REVIEW

File: db.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+cd /Users/kennedy.batista/dev/datomic-pro-1.0.6362
```
Comment (rmacario) [inline db.sh]: Dá pra deixar um caminho relativo aqui ?

**Review by rmacario — APPROVED** [review]

File: src/inflation/adapter/load_files.clj [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+(defn insert-entity [entity]
+  (prn entity)
+  (d/transact conn [entity]))
+
+(defn read-file [filename] 
+  (with-open [reader (io/reader filename)]
+    (doall
+     (csv/read-csv reader))))
+
+(defn create-price [item, country, type]
```
Comment (rfhayashi) [inline src/inflation/adapter/load_files.clj]: You can use destructuring to get the parts of the item:
```clojure
[[year-str month-str value-str] country]
```

File: src/inflation/adapter/load_files.clj [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+(defn load-file [filename country type]
+  (doseq [item (read-file filename)]
+    (-> item 
+        (first)
+        (str/split #";")
+        (create-price country type)
+        (insert-entity))))
+
+(load-file "resources/br_salary.csv" "BR" "SALARY")
+(load-file "resources/br_basic_food_basket.csv" "BR" "BFB")
```
Comment (rfhayashi) [inline src/inflation/adapter/load_files.clj]: Avoid keeping side effect things as top level forms. They will be run every time you evaluate the namespace. Prefer creating a function that is called when the application is started. If this is only for testing, wrap it in a `comment` form. Then you can run in the repl only when you need.

Comment (rfhayashi) [inline src/inflation/adapter/load_files.clj:38]: Avoid keeping side effect things as top level forms. They will be run every time you evaluate the namespace. Prefer creating a function that is called when the application is started. If this is only for testing, wrap it in a `comment` form. Then you can run in the repl only when you need.

File: src/inflation/adapter/load_files.clj [hunk 4]
```diff
@@ -0,0 +1,38 @@
+(ns inflation.adapter.load_files
+  (:require
+   [clojure.string :as str]
+   [datomic.api :as d]
+   [inflation.db.datomic.db :as db]
+   [inflation.model.price :as model]
+   [clojure.data.csv :as csv]
+   [clojure.java.io :as io]))
+
+(def conn (db/connection))
```
Comment (rfhayashi) [inline src/inflation/adapter/load_files.clj]: Avoid using vars to handle state. Prefer passing it as parameter to the functions and create the instance at application start up. If you start to have more stuff, consider using something like component (https://github.com/stuartsierra/component).

Comment (rfhayashi) [inline src/inflation/adapter/load_files.clj:10]: Avoid using vars to handle state. Prefer passing it as parameter to the functions and create the instance at application start up. If you start to have more stuff, consider using something like component (https://github.com/stuartsierra/component).

File: src/inflation/adapter/load_files.clj [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+(defn read-file [filename] 
+  (with-open [reader (io/reader filename)]
+    (doall
+     (csv/read-csv reader))))
+
+(defn create-price [item, country, type]
+  (let [value (bigdec (nth item 2))
+        month (long (Float/valueOf (nth item 1)))
+        year (long (Float/valueOf (nth item 0)))]
+    (model/new-price country type value year month)))
```
Comment (rfhayashi) [inline src/inflation/adapter/load_files.clj:27]: This is not only for clojure, but avoid having functions with several positional parameters. Those functions are hard to extend when you need new data. Prefer using a map instead.

File: src/inflation/model/price.clj [hunk 6]
```diff
@@ -0,0 +1,8 @@
+(ns inflation.model.price)
+
+(defn new-price [country type value year month]
```
Comment (rfhayashi) [inline src/inflation/model/price.clj:3]: Avoid functions that have several positional parameters, they are hard to extend and make the client code difficult to read. In this case the client code can just create the map (and you don't need this function).

Comment (rfhayashi) [inline src/inflation/model/price.clj:3]: Avoid functions that have several positional parameters, they are hard to extend and make the client code difficult to read. In this case the client code can just create the map (and you don't need this function).

File: src/inflation/db/datomic/db.clj [hunk 7]
```diff
@@ -0,0 +1,41 @@
+(ns inflation.db.datomic.db
+  (:require [datomic.api :as d]))
+
+(def db-uri "datomic:dev://localhost:4334/inflation")
```
Comment (rfhayashi) [inline src/inflation/db/datomic/db.clj]: Unless you do want to create a transactor to test how it is done, you can use datomic in memory (e.g. "datomic:mem://inflation").

Comment (rfhayashi) [inline src/inflation/db/datomic/db.clj:4]: Unless you do want to create a transactor to test how it is done, you can use datomic in memory (e.g. "datomic:mem://inflation").

**Review by rfhayashi — COMMENTED** [review]
Nice work! Made a few suggestions. Happy to have another look as the code evolves.

File: src/inflation/adapter/load_files.clj (hunk 4 above)
Comment (k-batista) [reply src/inflation/adapter/load_files.clj]: Yes, I made a 1:1 today and my friend commented to me about this library.
I need to understand how to always get the new datomic instance, so I don't have problems with updates in the db

File: src/inflation/db/datomic/db.clj (hunk 7 above)
Comment (k-batista) [reply src/inflation/db/datomic/db.clj]: I didn't know this, I will try

File: src/inflation/model/price.clj (hunk 6 above)
Comment (k-batista) [reply src/inflation/model/price.clj:3]: I understand your position, but for me still when I use a map as a parameter, I am forcing the client to know how to build the map and this is a dilemma for me. 
How can I make the client know how he should create the map? @rfhayashi

Comment (k-batista) [reply src/inflation/model/price.clj:3]: ~ thoughts java 🤣

Comment (rfhayashi) [reply src/inflation/model/price.clj:3]: It depends a lot on the context. In your case, the datomic schema tells that (https://github.com/k-batista/inflation/pull/1/files#diff-9cf35d4eb880e1cb7b2a7ce77a7645311fa5e362ad18d3ddff1dafd7c36c0363R13). Specs/schemas can help as well (https://clojure.org/guides/spec / https://github.com/plumatic/schema), although I'm not a super fan to use them in function level.

File: db.sh (hunk 1 above)
Comment (k-batista) [reply db.sh]: Da sim

**Pull request merged**